lib/modem_info/modem_info.c

```diff
@@ -983,9 +978,27 @@ int modem_info_get_operator(char *buf, size_t len)
 	//                                            |           |
 	//                                          start        end
 	const char comma = ',';
-	char *start = strchr(response, comma) + 1;
-	start = strchr(start, comma) + 2;
-	char *end = strchr(start, comma) - 2;
+	char *start = response;
+	for (size_t i = 0; i < XMONITOR_CMD_SHORT_OPERATOR_IDX; ++i) {
```

Comment:
Hmmm, maybe simpler to use sscanf?

Reply:
or maybe strtok is better
https://godbolt.org/z/E19Tejf4o

Reply:
Good point, this was not succinct. I like yours but I actually do think sscanf is more straightforward. Went forward with it. Open to feedback!